core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,21 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    public void setViews(Collection<View> views) throws IOException {
```

Comment:
Wait, why did you remove the `@DataBoundSetter` here? You need that for `DescribableModel` to recognize `views` as a mutable property.